lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -3374,11 +3399,22 @@ Error glow::optimizeFunction(Function *F, const Backend &B,
   // Optimize the graph again now that we have a lowered representation.
   ::glow::optimize(F, cctx);
 
+  // If requested fold ElemKind conversion Nodes into static Placeholders.
+  if (cctx.optimizationOpts.foldStaticPlaceholderConversions) {
+    FunctionPassManager FPM("FoldElemKindConversionIntoStaticPlaceholder",
+                            {FunctionPassID::FoldElemKindConversionIntoInputs});
+    if (FPM.run(F, cctx)) {
+      ::glow::optimize(F, cctx);
+    }
+  }
+
   // If requested, fold ElemKind conversion Nodes into inputs and outputs
   // (Placeholders and SaveNodes).
   if (cctx.optimizationOpts.foldElemKindConversionIntoIO) {
-    FunctionPassManager FPM("FoldElemKindConversionIntoIO",
-                            {FunctionPassID::FoldElemKindConversionIntoIO});
+    FunctionPassManager FPM(
+        "FoldElemKindConversionIntoIO",
+        {{FunctionPassID::FoldElemKindConversionIntoInputs},
```

Comment:
small nit: Looks a little funny/inefficient to me to back to back potentially create pipelines with `FoldElemKindConversionIntoInputs`, especially since they could both be run.

One thing you could do is merge this into a single `FunctionPassManager` -- check if `foldStaticPlaceholderConversions || foldElemKindConversionIntoIO`, create a `FunctionPassPipeline` with `FoldElemKindConversionIntoInputs` since it'll always be run, and then conditionally add `FoldElemKindConversionIntoOutputs` if `foldElemKindConversionIntoIO`. Then feed that `FunctionPassPipeline` into an `FPM`.

(whew that's a mouthful)